### DISCUSSION TITLE
Quote and CsvMode.NoEscape deliever unexpected result

### DISCUSSION THREAD

**Discussion by Rand-Random** [JoshClose/CsvHelper#2222] [Q&A] [answered]
Is it possible to read those values? 

```
1;""a""
2;"a;b;"
3;"a "b""
```

Expected result would be

| Col 1 | Col 2 |
| - | - |
| 1 | a |
| 2 | a;b; |
| 3 | a "b" |

I struggle to find the correct setting to support this file.
I believed that this would give me the result expected

```
var csvConfiguration = new CsvConfiguration(CultureInfo.CurrentCulture)
{
    Delimiter = ";",
    Quote = '\"',
    Mode = CsvMode.NoEscape
};
```

Which I expected to result into 
1) fields can be quoted eg. `"a;b;"`  --> so 1 field with value `a;b;`
2) fields don't need to escape quotes eg. `""a""` so 1 field with value `"a"`

It would be sufficient to get this result, and I would trim the quote `"` myself

| Col 1 | Col 2 |
| - | - |
| 1 | "a" |
| 2 | a;b; |
| 3 | a "b" |

**Comment by JoshClose** [comment, marked as answer]
`NoEscape` means there is no character escaping. This means you wouldn't be able to have a delimiter or line ending in a field. This isn't what you want.

`Escape` means you'll have an escape character that will precede a character you want to escape. If your escape character is `"`, a field that contained delimiter would like like this. `a";b` and would be output as `a;b`. This isn't what you want either.

`RFC4180` is the other and default mode. This uses rules that adhere to RFC 4180. That means fields that contain a delimiter, quote, or line ending, will need to be enclosed in `"field"` and any quote inside the field needs to be escaped with a quote `"field "" with quote"`.

What you're trying to do doesn't fall into any of these modes.

If you aren't escaping a quote inside a field, I don't see how you could determine that the field ends.
